polaris-core/src/main/java/org/apache/polaris/core/storage/FileStorageConfigurationInfo.java

```diff
@@ -42,7 +43,8 @@ public String getFileIoImplClassName() {
 
   @Override
   public void validatePrefixForStorageType(String loc) {
-    if (!loc.startsWith(getStorageType().getPrefix())
+    if (getStorageType().getPrefixes().stream()
```

Comment:
Given that we have the kind of weird-looking `    return storageType.getPrefixes().get(0) + bucket + "/" + keyPrefix;` in the `AwsCredentialsStorageIntegrationTest` I wonder if we should take this opportunity to get rid of `getPrefix()` and `getPrefixes()` and be more explicit about intent.

For this use case here it seems we could push down a method in `StorageType` like `isPathValidForStorageType` where we pass in the `loc.toLowerCase(Locale.ROOT)` here or just pass in `loc` and the `StorageType` itself can hold this logic.

And then we force ourselves to decide what we really meant in the integration test. The `.get(0)` "happens" to work for S3, but would've behaved weirdly if used with Azure.

We can choose one of these options there:

1. Just hard-code the construction of the path "s3 + ..." - it doesn't look like that integration test is really relying on being able to delegate out the prefix from StorageType
2. Really entrench the idea of a "canonical" scheme for a StorageType and a method like `getSchemePrefix` in `StorageType`, and then we should make `abfss://` the canonical one for Azure either by making that the first element of the list or even having a separate member that stores the canonical one.
3. Expose a helper method in `StorageType` like `constructAbsolutePathFromSchemelessPath` that abstracts away the choice of the canonical scheme.

Reply:
> For this use case here it seems we could push down a method in StorageType like isPathValidForStorageType where we pass in the loc.toLowerCase(Locale.ROOT) here or just pass in loc and the StorageType itself can hold this logic.

This is my preference; for AzureLocation we already have a method like `isAzurePath` but since we aren't actually implementing `AwsLocation` etc. here I didn't want to add that to the `StorageLocation` interface if unnecessary. 

For now, I'll hardcode the path into that test but let me add a comment to track the above